### ISSUE TITLE
Trigger animation more than once

### ISSUE THREAD

**Issue by nolanw** [MengTo/Spring#37]
Is there any way to run an animation more than once? I see that the spring resets itself after I call any `animate` method, so if I call `animate` again nothing happens. I'd like to apply whatever animation is set in Interface Builder each time I call `animate`.

**Comment by nolanw** [comment]
On reflection this might be a case of the XY problem, so I'll say what I'm actually trying to do.

I want to apply an animation (specified in IB) to a button each time a scroll view bounces.

I could record the animation parameters when the button initially loads, then reapply them each time I'd like to trigger the animation, but I'm hopeful there's another way.

**Comment by willowcheng** [comment]
@nolanw I'm facing the same problem for using animation more than once.

**Comment by MrMatthias** [comment]
without changing the code it's not possible. You could create a property  that you check in the resetAll method of the Spring class
```
func resetAll() {
    if !resetAfterAnimation {
        return
    }
    ...
}
```

**Comment by schneiderandre** [comment]
After an animation is complete, all related information are reset.
I'm not sure if these methods can be entirely deleted - I'm not aware of all side effects.

Maybe @MengTo knows if `reset`and `resetAll()` are needed, and if they can be removed.

**Comment by Aymenworks** [comment]
I encountered the same problem using storyboard, but I found a solution :+1:

Before calling animate() function, you need to set the animation property

```swift
self.newProfileNameTextField.animation = "shake"
self.newProfileNameTextField.animate()
```

**Comment by Mustafa-Altameemi** [comment]
@Aymenworks that's true .. that's true .. however, sometimes it works and sometimes it doesn't.
